### PR TITLE
Add masked and NaN means

### PR DESCRIPTION
**Pull request by qbarthelemy**
This PR aims to:
- add the masked Riemannian mean, for semi-orthogonal masks;
- add the Riemannian NaN-mean, a special case of masked Riemannian mean applied to covariance matrices corrupted by symmetric NaN values (in this case, masks are automatically estimated);
- improve `make_covariances`, ensuring positive eigen values and orthogonal eigen vectors (solve #19 ?);
- add a simple example comparing Euclidean NaN-mean, Riemannian NaN-mean and Riemannian mean after matrix deletion;
- complete tests.

F. Yger, S. Chevallier, Q. Barthélemy, S. Sra. "Geodesically-convex optimization for averaging partially observed covariance matrices", ACML 2021, [PDF](http://proceedings.mlr.press/v129/yger20a/yger20a.pdf)

Disclaimer: this is a pymanopt-free re-implementation of the original work coded with @sylvchev and @ygerf .

For comparison with your method: @ahippert

### CODE REVIEW

File: tests/test_utils_mean.py [hunk 1]
```diff
@@ -64,45 +67,76 @@ def test_riemann_mean(init, get_covmats_params):
...
+
+
+@pytest.mark.parametrize("init", [True, False])
+def test_riemann_mean_nan_shape(init, get_covmats):
+    """Test the riemann nan mean"""
+    n_matrices, n_channels = 10, 6
+    covmats = get_covmats(n_matrices, n_channels)
+    emean = np.mean(covmats, axis=0)
+    for i in range(n_matrices):
+        corrup_channels = np.random.randint(n_channels, size=n_channels // 2)
```
Comment (qbarthelemy) [inline tests/test_utils_mean.py:129]: @sylvchev , how can we use the reproducible random seed here? because it seems not explicitely available...

Comment (sylvchev) [reply tests/test_utils_mean.py:129]: there's a fixture for that ©️  I'll push the modification.

**Review by sylvchev — APPROVED** [review]

**Comment by sylvchev** [comment]
I proposed a new plot, mimicking one of the figure from the article. It is just a suggestion, you could remove it.

**Comment by qbarthelemy** [comment]
> I proposed a new plot, mimicking one of the figure from the article. It is just a suggestion, you could remove it.

The new plot is great! Thanks!

**Comment by sylvchev** [comment]
Ok, LGTM!
Thanks @qbarthelemy 🙏

**Pull request merged**